e2e/testsuite/semver.go

```diff
@@ -0,0 +1,75 @@
+package testsuite
+
+import (
+	"strconv"
+	"strings"
+)
+
+// Version follows semantic versioning while disregarding the patch version.
+type Version struct {
+	Major uint64
+	Minor uint64
+}
+
+// NewVersion constructs a new version given the major and minor version.
+func NewVersion(major, minor uint64) Version {
+	return Version{
+		Major: major,
+		Minor: minor,
+	}
+}
+
+func ParseChainVersion(version string) (Version, bool) {
+	// remove any "v" label
+	version = strings.TrimPrefix(version, "v")
+
+	versions := strings.Split(version, ".")
+	if len(versions) != 3 {
+		return Version{}, false
+	}
+
+	major, err := strconv.ParseUint(versions[0], 10, 64)
+	if err != nil {
+		return Version{}, false
+	}
+
+	minor, err := strconv.ParseUint(versions[1], 10, 64)
+	if err != nil {
+		return Version{}, false
+	}
+
+	return Version{
+		Major: major,
+		Minor: minor,
+	}, true
```

Comment:
The majority of the parsing logic here seems to be duplicating  `semver.Major` and `semver.Minor`

Reply:
ah! I was looking for that, wasn't sure it existed, looks to be importable under "semver"?